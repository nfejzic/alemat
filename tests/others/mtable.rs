use alemat::{
    elements::{ColumnLine, Frac, Ident, Num, Operator, TableAttr},
    MathMl, MathMlFormatter,
};

#[test]
fn matrix() {
    let out = MathMl::with_content(alemat::children![
        Frac::builder()
            .num(Ident::from("A"))
            .denom(Num::from(2))
            .build(),
        Operator::eq(),
        alemat::row![
            Operator::lparens(),
            alemat::table![
                [Num::from(1), Num::from(2), Num::from(3)],
                [Num::from(4), Num::from(5), Num::from(6)],
                [Num::from(7), Num::from(8), Num::from(9)],
            ],
            Operator::rparens(),
        ]
    ])
    .render(&mut MathMlFormatter);

    crate::snap_test!(out, name: "others_table_matrix");
}

#[test]
fn matrix_augmented() {
    let out = MathMl::with_content(alemat::children![
        Frac::builder()
            .num(Ident::from("A"))
            .denom(Num::from(2))
            .build(),
        Operator::eq(),
        alemat::row![
            Operator::lparens(),
            alemat::table![
                [Num::from(1), Num::from(2), Num::from(3)],
                [Num::from(4), Num::from(5), Num::from(6)],
                [Num::from(7), Num::from(8), Num::from(9)],
            ]
            .with_attr([TableAttr::ColumnLines(vec![
                ColumnLine::None,
                ColumnLine::Solid,
                ColumnLine::None
            ])]),
            Operator::rparens(),
        ]
    ])
    .render(&mut MathMlFormatter);

    crate::snap_test!(out, name: "others_table_matrix_augmented");
}

#[test]
fn matrix_square() {
    let out = MathMl::with_content(alemat::children![
        Frac::builder()
            .num(Ident::from("A"))
            .denom(Num::from(2))
            .build(),
        Operator::eq(),
        alemat::row![
            Operator::lbracket(),
            alemat::table![
                [Num::from(1), Num::from(2), Num::from(3)],
                [Num::from(4), Num::from(5), Num::from(6)],
                [Num::from(7), Num::from(8), Num::from(9)],
            ],
            Operator::rbracket(),
        ]
    ])
    .render(&mut MathMlFormatter);

    crate::snap_test!(out, name: "others_table_matrix_square");
}
