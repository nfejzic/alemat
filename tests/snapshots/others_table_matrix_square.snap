---
source: tests/others/mtable.rs
expression: input
---
<math>
  <mfrac>
    <mi>
      A
    </mi>
    <mn>
      2
    </mn>
  </mfrac>
  <mo>
    =
  </mo>
  <mrow>
    <mo>
      [
    </mo>
    <mtable>
      <mtr>
        <mtd>
          <mn>
            1
          </mn>
        </mtd>
        <mtd>
          <mn>
            2
          </mn>
        </mtd>
        <mtd>
          <mn>
            3
          </mn>
        </mtd>
      </mtr>
      <mtr>
        <mtd>
          <mn>
            4
          </mn>
        </mtd>
        <mtd>
          <mn>
            5
          </mn>
        </mtd>
        <mtd>
          <mn>
            6
          </mn>
        </mtd>
      </mtr>
      <mtr>
        <mtd>
          <mn>
            7
          </mn>
        </mtd>
        <mtd>
          <mn>
            8
          </mn>
        </mtd>
        <mtd>
          <mn>
            9
          </mn>
        </mtd>
      </mtr>
    </mtable>
    <mo>
      ]
    </mo>
  </mrow>
</math>

